src/profile/sigmapr.jl

```diff
@@ -0,0 +1,67 @@
+"""
+    refitσ!(m::LinearMixedModel{T}, σ::T, tc::TableColumns{T}, obj::T, neg::Bool)
+
+Refit the model `m` with the given value of `σ` and return a NamedTuple of information about the fit.
+
+`obj` and `neg` allow for conversion of the objective to the `ζ` scale and `tc` is used to return a NamedTuple
+"""
+function refitσ!(
+    m::LinearMixedModel{T}, σ, tc::TableColumns{T}, obj::T, neg::Bool
+) where {T}
+    m.optsum.sigma = σ
+    refit!(m; progress=false)
+    return mkrow!(tc, m, (neg ? -one(T) : one(T)) * sqrt(m.objective - obj))
+end
+
+"""
+    _facsz(m, σ, objective)
+
+Return a factor such that refitting `m` with `σ` at its current value times this factor gives `ζ ≈ 0.5`
+"""
+function _facsz(m::LinearMixedModel{T}, σ::T, obj::T) where {T}
+    i64 = inv(64)
+    m.optsum.sigma = σ * exp(i64)
```

Comment:
I'm guessing that LLVM will generally catch that `exp(i64)` is const and generate ASM appropriately, but would it make sense to move that out of the multiplication?

Reply:
I will move the evaluation of `exp` to a separate assignment, which should help.

Reply:
I checked with @code_warntype and the compiler does infer that `expi64` in the revised code is constant.